### ISSUE TITLE
pad_col in DeepHit pmf

### ISSUE THREAD

**Issue by thbuerg** [havakv/pycox#47]
Hi Havard,

I do have a question regarding the pmf computation in the DeepHit Models: As I noticed, you are zero-padding the logits at the end before taking the softmax and I was wondering what the reason behind that was? Why is it not enough to take the softmax on the unpadded logits?

Many thanks for your help & time!

**Comment by havakv** [comment]
Hi Thore,

This is a good question, with multiple parts to the answer, so let us start by only considering the DeepHitSingle model (without competing risk) for simplicity.

First, note that DeepHit is slightly different in pycox than how the original authors described/implemented it. In [the paper](http://medianetlab.ee.ucla.edu/papers/AAAI_2018_DeepHit) the consider a fixed maximum time of survival T_max, meaning no individual can survive past T_max. It is quite uncommon in survival analysis, and we don't see any benefits of the restriction (other than simpler implementation) so we removed it. The padding represents the survival past time T_max.

Second, we wanted to allow survival past T_max without adding an extra discretization point, as this allow the number of output nodes in the nework to be the same in both discrete hazard methods (LogisticHazard NnetSurvival) as in the PMF methods such as DeepHit. To do this, we simply use the statistical convention of a softmax with a reference class as described in equation (9) in [this paper](https://arxiv.org/pdf/1910.06724.pdf). This is why we pad with the value zero.

If you only remove the zero padding, that would be to set the probability of survival past time T_max to zero. This can be fine, but then you need to make sure you have a sufficiently high T_max, to ensure this assumption is reasonable. 

The same argument follows for the competing risks model for DeepHIt.

Does this answer your question?

**Comment by thbuerg** [comment]
Oh, this makes a lot of sense indeed. Thank you for your kind and detailed answer, I should have gone through your paper more carefully..

Thanks!!

**Comment by havakv** [comment]
No worries. It would require *really* careful reading of the papers to understand this, so it makes sense to just ask instead. I'm sure you're not the last one wondering about this, so it makes sense to have it written here.

Happy to help!
